src/Disks/DiskLocal.cpp

```diff
@@ -292,6 +296,28 @@ void DiskLocal::copy(const String & from_path, const std::shared_ptr<IDisk> & to
         IDisk::copy(from_path, to_disk, to_path); /// Copy files through buffers.
 }
 
+int DiskLocal::open(const String & path, mode_t mode) const
+{
+    String full_path = disk_path + path;
+    int fd = ::open(full_path.c_str(), mode);
+    if (-1 == fd)
+        throwFromErrnoWithPath("Cannot open file " + full_path, full_path,
+                        errno == ENOENT ? ErrorCodes::FILE_DOESNT_EXIST : ErrorCodes::CANNOT_OPEN_FILE);
+    return fd;
+}
+
+void DiskLocal::close(int fd) const
+{
+    if (-1 == ::close(fd))
+        throw Exception("Cannot close file", ErrorCodes::CANNOT_CLOSE_FILE);
+}
+
+void DiskLocal::sync(int fd) const
+{
+    if (-1 == ::fsync(fd))
```

Comment:
`fdatasync()` is twice faster sometimes(!) It's much better to use it here if supported.

Reply:
Also, it's better to fix other places. I mean replace fsync with fdatasync

Reply:
Also, I'm curious in details - does CH fsyncs in the background (in separate thread)?

Reply:
It was already replaced to `fdatasync` recently #31229

Reply:
> Also, I'm curious in details - does CH fsync in the background (in separate thread)?

No, fsync is done in the same thread.

Reply:
I think it's better to fsync in the background. Possibly using fsync_data_range in order not to fsync newly arrived data